File: 백준/Bronze/13410. 거꾸로 구구단/거꾸로 구구단.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        int max = 0;

        for (int i = 1; i < k + 1; i++) {
            int temp = n * i;
            int chk = 0;
            while (temp != 0) {
                chk = 10 * chk + temp % 10;
                temp = temp / 10;
            }
            max = Math.max(max, chk);
        }
        System.out.println(max);
    }
}

File: 백준/Bronze/13410. 거꾸로 구구단/거꾸로 구구단.py
n, k = map(int, input().split())
maximum = 0

for i in range(1, k + 1):
    temp = n * i
    chk = 0
    while temp:
        chk = 10 * chk + temp % 10
        temp //= 10
    maximum = max(maximum, chk)

print(maximum)